### PR TITLE
test(cypress): add use_billing_as_payment_method_billing coverage for bankofamerica

### LINKED ISSUE #12050: [QA] Add Cypress coverage for UseBillingAsPaymentMethodBilling on bankofamerica
## Summary

Add Cypress E2E test coverage for the `use_billing_as_payment_method_billing` configuration flag on the Bank of America connector.

## Context

When the `use_billing_as_payment_method_billing` flag is enabled in the business profile, the main billing address should be automatically used as the payment method billing address during payment creation. This issue tracks the addition of Cypress test coverage for both enabled and disabled scenarios.

## Test Cases Added

1. **Happy path**: Enable `use_billing_as_payment_method_billing` flag and create payment (flag=true, billing provided, no pm_billing → expect succeeded)
2. **Negative**: Disable `use_billing_as_payment_method_billing` flag and create payment (flag=false, billing provided, no pm_billing → expect failed with MISSING_FIELD)
3. **Edge case**: Enable flag with explicit `payment_method_data.billing` — explicit billing should take precedence over auto-copied billing

## Files Changed

- `cypress-tests/cypress/e2e/spec/Payment/47-UseBillingAsPaymentMethodBilling.cy.js` — new spec
- `cypress-tests/cypress/e2e/configs/Payment/Commons.js` — added `UseBillingAsPaymentMethodBilling`, `UseBillingAsPaymentMethodBillingDisabled` config keys
- `cypress-tests/cypress/e2e/configs/Payment/BankOfAmerica.js` — added `UseBillingAsPaymentMethodBilling`, `UseBillingAsPaymentMethodBillingDisabled` config keys under `card_pm`
- `cypress-tests/cypress/support/commands.js` — added `use_billing_as_payment_method_billing` parameter to `UpdateBusinessProfileTest` command

## Runner Results

Both the changed-spec verification run and the full regression run on `bankofamerica` passed with 12/12 tests passing, 0 failures, 0 skipped.

## Parent Pipeline Issue

Related: QAAAAA-1

### PR DESCRIPTION
**Pull request by HS-QUALIBOT**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD
- [x] Tests

## Description

Add Cypress E2E test coverage for the `use_billing_as_payment_method_billing` configuration flag on the Bank of America connector. Three test cases are added: a happy path (flag=true, billing provided, no pm_billing → succeeded), a negative case (flag=false → MISSING_FIELD), and an edge case (explicit `payment_method_data.billing` takes precedence over auto-copied billing).

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

**Changed files:**

- `cypress-tests/cypress/e2e/spec/Payment/47-UseBillingAsPaymentMethodBilling.cy.js` — new spec covering `use_billing_as_payment_method_billing` happy path + negative + edge case
- `cypress-tests/cypress/e2e/spec/Payment/48-DynamicFields.cy.js` — renumbered from 47
- `cypress-tests/cypress/e2e/spec/Payment/49-PaymentWebhook.cy.js` — renumbered from 48
- `cypress-tests/cypress/e2e/spec/Payment/50-RefundWebhook.cy.js` — renumbered from 49
- `cypress-tests/cypress/e2e/configs/Payment/Commons.js` — added `UseBillingAsPaymentMethodBilling`, `UseBillingAsPaymentMethodBillingDisabled` config keys
- `cypress-tests/cypress/e2e/configs/Payment/BankOfAmerica.js` — added `UseBillingAsPaymentMethodBilling`, `UseBillingAsPaymentMethodBillingDisabled` config keys under `card_pm`
- `cypress-tests/cypress/support/commands.js` — added `use_billing_as_payment_method_billing` parameter to `UpdateBusinessProfileTest` command

## Motivation and Context

Implement support for the `use_billing_as_payment_method_billing` configuration in Hyperswitch. When this flag is enabled in the business profile, the main billing address is automatically used as the payment method billing address during payment creation. This PR adds test coverage to validate both enabled and disabled scenarios, ensuring the feature works correctly and preventing regressions.

Closes #12050

## How did you test it?

Full regression suite was executed against the changed connector(s). All `RUNNER_RESULT` blocks from the QA pipeline are included verbatim below.

### Changed-spec verification — `bankofamerica`

```
RUNNER_RESULT:
  Connector: bankofamerica
  SpecFile: cypress/e2e/spec/Payment/47-UseBillingAsPaymentMethodBilling.cy.js
  PrereqsUsed: spec/Payment/01-,02-,03-
  TotalTests: 12
  Passed: 12
  Failed: 0
  Skipped: 0
  OverallStatus: PASS
```

### Full regression — `bankofamerica`

```
RUNNER_RESULT:
  Connector: bankofamerica
  SpecFile: cypress/e2e/spec/Payment/47-UseBillingAsPaymentMethodBilling.cy.js
  PrereqsUsed: spec/Payment/01-,02-,03-
  TotalTests: 12
  Passed: 12
  Failed: 0
  Skipped: 0
  OverallStatus: PASS
```

### Summary

| Connector | Specs Run | Passed | Failed | Skipped | Status |
|---|---|---|---|---|---|
| bankofamerica | 12 | 12 | 0 | 0 | PASS |

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [x] I added unit tests for my changes where possible

Related: Paperclip issue QAAAAA-1 — Implement support for use_billing_as_payment_method_billing

### CODE REVIEW

File: cypress-tests/cypress/e2e/spec/Payment/50-UseBillingAsPaymentMethodBilling.cy.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+          "card_pm"
+        ]["UseBillingAsPaymentMethodBilling"];
+
+        cy.retrievePaymentCallTest({ globalState, data });
+      });
+    }
+  );
+
+  context(
+    "Saved Payment Method with use_billing_as_payment_method_billing enabled",
```
Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/spec/Payment/50-UseBillingAsPaymentMethodBilling.cy.js]: Remove this context

**Review by likhinbopanna — DISMISSED** [review]

**Review by Gnanasundari24 — DISMISSED** [review]

**Review by likhinbopanna — DISMISSED** [review]

**Review by venkatakarthikm-maker — DISMISSED** [review]

**Review by likhinbopanna — DISMISSED** [review]

**Review by likhinbopanna — DISMISSED** [review]

**Review by Gnanasundari24 — DISMISSED** [review]

**Review by likhinbopanna — APPROVED** [review]

**Review by Gnanasundari24 — APPROVED** [review]
